docs/wiki/concepts/Historical-Data.md

```diff
@@ -0,0 +1,146 @@
+## Table of Contents
+
+- [Table of Contents](#table-of-contents)
+- [Historical Buffers](#historical-buffers)
+- [Historical Range Access](#historical-range-access)
+
+## Historical Buffers
+
+`csp` provides access to historical input data as well.
+By default only the last value of an input is kept in memory, however one can request history to be kept on an input either by number of ticks or by time using **csp.set_buffering_policy.**
+
+The methods **csp.value_at**, **csp.time_at** and **csp.item_at** can be used to retrieve historical input values.
+Each node should call **csp.set_buffering_policy** to make sure that its inputs are configured to store sufficiently long history for correct implementation.
+For example, let's assume that we have a stream of data and we want to create equally sized buckets from the data.
+A possible implementation of such a node would be:
+
+```python
+@csp.node
+def data_bin_generator(bin_size: int, input: ts['T']) -> ts[['T']]:
+    with csp.start():
+        assert bin_size > 0
+        # This makes sure that input stores at least bin_size entries
+        csp.set_buffering_policy(input, tick_count=bin_size)
+    if csp.ticked(input) and (csp.num_ticks(input) % bin_size == 0):
+        return [csp.value_at(input, -i) for i in range(bin_size)]
+```
+
+In this example, we use **`csp.set_buffering_policy(input, tick_count=bin_size)`** to ensure that the buffer history contains at least **`bin_size`** elements.
+Note that an input can be shared by multiple nodes, if multiple nodes provide size requirements, the buffer size would be resolved to the maximum size to support all requests.
+
+Alternatively, **`csp.set_buffering_policy`** supports a **`timedelta`** parameter **`tick_history`** instead of **`tick_count`.**
+If **`tick_history`** is provided, the buffer will scale dynamically to ensure that any period of length **`tick_history`** will fit into the history buffer.
+
+To identify when there are enough samples to construct a bin we use **`csp.num_ticks(input) % bin_size == 0`**.
+The function **`csp.num_ticks`** returns the number or total ticks for a given time series.
+NOTE: The actual size of the history buffer is usually less than **`csp.num_ticks`** as buffer is dynamically truncated to satisfy the set policy.
+
+The past values in this example are accessed using **`csp.value_at`**.
+The various historical access methods take the same arguments and return the value, time and tuple of `(time,value)` respectively:
+
+- **`csp.value_at`**`(ts, index_or_time, duplicate_policy=DuplicatePolicy.LAST_VALUE, default=UNSET)`: returns **value** at requested `index_or_time`
```

Comment:
lets add " of the timeseries" to these three lines

Reply:
Let me know if this is what you had in mind.

Reply:
Yes, this looks right, thanks!